onboarding_auth/onboardingAuthData/src/main/java/com/alexafit/onboardingauthdata/repository/OnboardingAuthRepositoryImpl.kt

```diff
@@ -2,21 +2,29 @@ package com.alexafit.onboardingauthdata.repository
 
 import com.alexafit.core.local.datastore.PreferenceStorage
 import com.alexafit.onboardingauthdata.mapper.mapToDto
+import com.alexafit.onboardingauthdata.model.remote.RegisterResponse
 import com.alexafit.onboardingauthdata.remote.TaskyApi
 import com.alexafit.onboardingauthdomain.model.remote.LoginUser
 import com.alexafit.onboardingauthdomain.model.remote.RegisterUser
 import com.alexafit.onboardingauthdomain.repository.OnboardingAuthRepository
+import com.squareup.moshi.Moshi
 
 class OnboardingAuthRepositoryImpl(
     private val taskyApi: TaskyApi,
     private val dataStore: PreferenceStorage
 ) : OnboardingAuthRepository {
-    override suspend fun registerUser(registerUser: RegisterUser): Result<String?> {
+    override suspend fun registerUser(registerUser: RegisterUser): Result<Unit> {
         val registerResponse = taskyApi.registerUser(registerDto = registerUser.mapToDto())
-        return if (registerResponse.isSuccessful) {
-            Result.success(null)
-        } else {
-            Result.failure(Throwable(message = registerResponse.body()?.message))
+        try {
+            return if (registerResponse.isSuccessful) {
+                Result.success(Unit)
+            } else {
+                val errorJson = registerResponse.errorBody()?.string()
+                val moshiAdapter = Moshi.Builder().build().adapter(RegisterResponse::class.java)
+                return Result.failure(Throwable(message = errorJson?.let { moshiAdapter.fromJson(it)?.message }))
+            }
+        } catch (e: Exception) {
+            return Result.failure(Throwable(message = "Error in api call"))
```

Comment:
Also, you'll probably have similar code like this quite often in the app, it could make sense to create a generic helper function for this which handles parsing the result into Kotlin's `Result` class